[[ Mac64 ]] -[NSWindow backingScaleFactor] sometimes returns NaN

livecodefraser opened the pull request:
I'm not entirely happy with having to do this as I'm surprised that NaN can ever be returned. Maybe we're doing something strange?

livecodefraser committed:
[[ Mac64 ]] -[NSWindow backingScaleFactor] sometimes returns NaN

runrevmark commented:
@livecodefraser: Yes - that is odd. If the selector didn't exist it would throw an exception I think... I can't see anything wrong with the code unless 'fpret' form of message sending isn't needed on x64?

runrevmark commented:
Methinks it might be because of this - http://stackoverflow.com/questions/8088145/running-performselector-on-an-object-that-returns-a-double-not-id

livecodefraser committed:
[[ Mac64 ]] Ensure we use objc_msgSend_fpret correctly

This method needs a function pointer cast in 64-bit mode in order
to behave correctly. This commit adds a template wrapping this
to make it a bit easier to use.

peter-b commented:
@livecode-vulcan review ok 4160e46

livecode-vulcan commented:
💙 review by @peter-b ok 4160e460f902b676c4bf7d53e7e00431370b2a1c

livecode-vulcan commented:
😎 test success 4160e460f902b676c4bf7d53e7e00431370b2a1c
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/1007)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/1015)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/1004)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/1007)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/1018)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/1025)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/1022)
